Add API Sever To TicTacToe

gilroman opened the pull request.

Commit:
Create API

tk-kowal commented on src/tic_tac_toe_pair/routes.clj:
```diff
@@ -0,0 +1,19 @@
+        (POST "/move" [] (ring-json/wrap-json-body move-handler))))
+
+    
+(defroutes non-body-routes 
```
Are these non-body-routes only needed for testing? I see one test where you're hitting the `/trouble` endpoint to verify your error handler is working as expected. Could you mock one of your handlers for a real route to force it to throw an exception instead?

gilroman replied:
Good point. Yes, I created that endpoint just to test the middleware that catches exceptions. I'll get rid of this endpoint and write a test that forces one of the real endpoints to throw an exception.

tk-kowal commented on test/tic_tac_toe_pair/handlers_test.clj:
```diff
@@ -0,0 +1,81 @@
+                              (mock/content-type "application/json")))]
+        (testing "returns a 200 status"
+          (is (= 200 (:status response)))
+        (testing "returns a valid game map"
```
It's a bit unexpected that the `/player` endpoint would return a game map as well. If you like the convenience of making one request to the backend to get a new game then consider changing your route to reflect that it returns more than just a player.

gilroman replied:
Thanks Tom. I'll rename the endpoint to game instead.

cmvandrevala commented on src/tic_tac_toe_pair/api.clj:
```diff
@@ -0,0 +1,9 @@
+(def app 
+  (-> api-routes
+      (wrap-defaults api-defaults )
+      wrap-500-catchall))
```
Editors usually have plugins that will automatically trim excess whitespace and add a newline to the bottom of every file you save. You might want to consider that for your editor of choice.

gilroman replied:
Thank you Cyrus. I will get these styling rules setup on VSCode.

cmvandrevala commented on test/tic_tac_toe_pair/middleware_test.clj:
```diff
@@ -0,0 +1,13 @@
+            [tic-tac-toe-pair.api :refer [app]]
+            [tic-tac-toe-pair.middleware :refer :all]))
+
+(deftest wrap-500-catchall-test 
```
@tk-kowal already mentioned this above, and I agree with his comment. It was pretty tough to grok this test since the `/trouble` endpoint was not implementing any TTT-related behavior. I would try mocking a handler for a real route instead of creating routes that are only used in the tests

gilroman replied:
Thanks for reviewing my code, Cyrus! I'm going to address this by changing the test for this middleware so that it hits a real endpoint related to the game, and will also get rid of the unused endpoint.

Commit:
Address Feedback

- Addesses feedback regarding unused endpoints
- Addresses feedback regarding trailing whitespace and ending newlinees
- Addresses feedback regarding renaming player endpoing
